### PR TITLE
fix: add Spanish locale in datepicker

### PR DESCRIPTION
**Pull request by JMMOLLER**
### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [x] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 💡 Background and solution

Additional properties were added for a more complete Spanish translation in the DatePicker component.

### 📝 Changelog

Properties were added in the date-picker for the es_ES locale.

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | add properties in es_ES |
| 🇨🇳 Chinese | 补充 西班牙语 多语言 |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/date-picker/locale/es_ES.ts [hunk 1]
```diff
@@ -8,9 +8,28 @@ const locale: PickerLocale = {
...
+      'Ago',
+      'Sep',
+      'Oct',
+      'Nov',
+      'Dic',
+    ],
     ...CalendarLocale,
   },
   timePickerLocale: {
+    placeholder: 'Seleccionar hora',
```
Comment (yoyo837) [inline components/date-picker/locale/es_ES.ts:32]: Move it into `components/time-picker/locale/es_ES.ts`

File: components/date-picker/locale/es_ES.ts [hunk 2]
```diff
@@ -8,9 +8,28 @@ const locale: PickerLocale = {
   lang: {
     placeholder: 'Seleccionar fecha',
     rangePlaceholder: ['Fecha inicial', 'Fecha final'],
+    shortWeekDays: ['Dom', 'Lun', 'Mar', 'Mie', 'Jue', 'Vie', 'Sab'],
+    yearFormat: 'YYYY',
```
Comment (yoyo837) [inline components/date-picker/locale/es_ES.ts:12]: Some fields may need to be moved to `rc-picker/lib/locale/es_ES`.

File: components/date-picker/locale/es_ES.ts (hunk 1 above)
Comment (JMMOLLER) [reply components/date-picker/locale/es_ES.ts]: I didn't realize that file was there. It's fixed now! [Here.](https://github.com/ant-design/ant-design/pull/48601/commits/7102467d410b0f7b70d96c7fe8e3a5613d20b69a)

**Review by zombieJ — DISMISSED** [review]

**Comment by zombieJ** [comment]
Test failed. Pls check to update the test case. If it's correct, use `npm test -- -u` to update snapshot.

**Review by li-jia-nan — APPROVED** [review]

**Pull request merged**